### ISSUE TITLE
Pre-check on special characters when login maybe too strict

### ISSUE THREAD

**Issue by Arsnael** [linagora/tmail-flutter#1929]
### Description

A new user from our company got his account created, with a pre-generated password. When trying to login via tmail in tmail.linagora.com, it was returning bad credentials. While he could connect without issues to the mail via IMAP (thunderbird) and OpenPaas.

After investigating a while, the only option I could see was potential special characters in the password that would be rejected right away by tmail-flutter app. After that the user changed his password, he could connect without issue.

The only thing I know is that the password contained two special characters:
- "|"
- "?"

Note that tmail.linagora.com is not using OIDC (yet), but normal authentication.

### Expected result

The user should not be rejected concerning those special characters.

### Current behavior

Tmail frontend rejects those (it is not the backend as Openpaas and IMAP thunderbird did not present such issue)

### Preconditions (optional)

Have a password with "|" and/or "?" special characters in it

#### Reproduction Steps

- Create a new user or change a user password with "|" and/or "?" 
- Try to login with normal authentication (not OIDC!!!)

### Context

Web browser, tmail.linagora.com

### Additional information

None

**Comment by QuangHoang1210** [comment]
@dab246  @hoangdat  @Arsnael 

**Expected:** 
`Mot de passe : Le mot de passe est invalide - de 4 à 12 caractères (alphanumériques ou -$\&~#{([|_`^@);]+=}%!:/.,?<>") !`

~~ Password: The password is invalid - 4 to 12 characters (alphanumeric or -$\&~#{([|_`^@);]+=}%!:/.,?<>") !

**Actual:**
I can change password which include special characters and it match condition above then I login successfully.

**Additional information:** I've try with the password requirement above and it's working well. 

**Example video**
https://github.com/linagora/tmail-flutter/assets/124866146/a1464050-7f52-4ab2-b86f-fc7d2d5d4f34

**Comment by Arsnael** [comment]
I honestly don't know then... You don't ask the password of someone in prod so I do not have that element unfortunately

**Comment by QuangHoang1210** [comment]
@Arsnael  yep, let me reproduce and investigate more about this issue. I will give you updates once I have news

**Comment by chibenwa** [comment]
IMO we should accept whatever string supplied by the user?
We are not creating a password here.
We MAY work with system with other pre-requisite than OBM...
